### PR TITLE
Test waitUntilFinish(Duration) in the DirectRunner

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
Ensures that the call to "waitUntilFinish(Duration)" terminates before
the Pipeline completes.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @iemejia

**Comment by iemejia** [comment]
I think the issue I reported on this is also probably related to the fact that I was not using setBlockOnRun(false), I didn't even know about this subtle difference that this does and that does not exist in other runners. I will test this and confirm you if everything is OK in this PR.

File: runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectRunnerTest.java [hunk 1]
```diff
@@ -277,6 +278,34 @@ public PipelineResult call() {
...
+                }));
+    PipelineResult result = p.run();
+    // The pipeline should never complete;
+    assertThat(result.getState(), is(State.RUNNING));
+    // Must time out, otherwise this test will never complete
+    result.waitUntilFinish(Duration.millis(1L));
+    assertThat(result.getState(), is(State.RUNNING));
+
+    result.cancel();
+    result.waitUntilFinish();
```
Comment (tgroh) [inline runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectRunnerTest.java]: It's not necessary. Done.

**Comment by iemejia** [comment]
LGTM.
(I will fix some small cosmetic issues before merging).

**Pull request closed without merging**